src/index.js

```diff
@@ -102,39 +132,47 @@ class Dropzone extends React.Component {
 
   onDrop(e) {
     const { onDrop, onDropAccepted, onDropRejected, multiple, disablePreview } = this.props;
-    const fileList = getDataTransferItems(e, multiple);
     const acceptedFiles = [];
     const rejectedFiles = [];
 
     // Stop default browser behavior
     e.preventDefault();
 
-    // Reset the counter along with the drag on a drop.
-    this.enterCounter = 0;
+    // Reset the target list along with the drag on a drop.
+    this.dragTargets = [];
     this.isFileDialogActive = false;
 
-    fileList.forEach((file) => {
-      if (!disablePreview) {
-        file.preview = window.URL.createObjectURL(file); // eslint-disable-line no-param-reassign
-      }
+    // Don't fire callbacks if a child Dropzone component has already handled this event
+    // eslint-disable-next-line no-underscore-dangle
+    if (!e.__dropzoneChildHandledThis) {
```

Comment:
OK. I don't immediately need nested dropzone support for my own use case yet, but might in the future. I think it might be surprising for someone who tries nesting to find that the parent still receives a drop event, but it's hard to predict whether anyone will actually want to do that. I'll go ahead and remove it from the 2 PR's and stash the changes somewhere; maybe it could be a separate PR by itself later if needed.

I tested the sibling case manually, and it works fine. A single user drop gets fired like:

```
event chain 1: sibling 1 document drop: <-- this isn't me, e.preventDefault() -->
event chain 2: sibling 2 document drop: <-- don't call e.preventDefault() -->
event chain 2: sibling 2 dropzone instance: <-- receives event -->
```